src/main/groovy/org/shipkit/internal/gradle/VersionUpgradeConsumerPlugin.java

```diff
@@ -9,170 +9,164 @@
 import org.shipkit.gradle.git.GitPushTask;
+import org.shipkit.internal.gradle.configuration.DeferredConfiguration;
 import org.shipkit.internal.gradle.versionupgrade.CreatePullRequestTask;
+import org.shipkit.internal.gradle.versionupgrade.DependencyNewVersionParser;
 import org.shipkit.internal.gradle.versionupgrade.ReplaceVersionTask;
-import org.shipkit.internal.gradle.configuration.LazyConfiguration;
 import org.shipkit.internal.gradle.configuration.ReleaseConfigurationPlugin;
 import org.shipkit.internal.gradle.git.GitCheckOutTask;
 import org.shipkit.internal.gradle.git.GitPush;
 import org.shipkit.internal.gradle.util.TaskMaker;
-import org.shipkit.internal.util.EnvVariables;
-import org.shipkit.internal.util.ExposedForTesting;
-
-import javax.inject.Inject;
 
 /**
  * BEWARE! This plugin is in incubating state, so its API may change in the future!
  * The plugin adds following tasks:
  *
  * <ul>
- *     <li>versionUpgradeCheckoutBaseBranch - checkouts base branch - the branch to which version upgrade should be applied through pull request</li>
- *     <li>versionUpgradeCheckoutVersionBranch - checkouts version branch - a new branch where version will be upgraded</li>
- *     <li>versionUpgradeReplaceVersion - replaces version in build file, using dependency pattern</li>
- *     <li>versionUpgradeGitCommit - commits replaced version</li>
- *     <li>versionUpgradeGitPush - pushes the commit to the version branch</li>
- *     <li>versionUpgradeCreatePullRequest - creates a pull request between base and version branches</li>
+ *     <li>checkoutVersionUpgradeBaseBranch - checkouts base branch - the branch to which version upgrade should be applied through pull request</li>
+ *     <li>checkoutVersionUpgradeVersionBranch - checkouts version branch - a new branch where version will be upgraded</li>
+ *     <li>replaceVersion - replaces version in build file, using dependency pattern</li>
+ *     <li>commitVersionUpgrade - commits replaced version</li>
+ *     <li>pushVersionUpgrade - pushes the commit to the version branch</li>
+ *     <li>createPullRequest - creates a pull request between base and version branches</li>
  *     <li>performVersionUpgrade - task aggregating all of the above</li>
  * </ul>
  *
  * Plugin should be used in client projects that want to have automated version upgrades of some other dependency, that use the producer version of this plugin.
  * Project with the producer plugin applied would then clone a fork of client project and run './gradlew performVersionUpgrade -PdependencyNewVersion=${VERSION}' on it.
+ *
+ * Example of plugin usage:
+ *
+ * Configure your 'shipkit.gradle' file like here:
+ *
+ *      apply plugin: 'org.shipkit.version-upgrade-consumer'
+ *
+ *      versionUpgrade{
+ *          baseBranch = 'release/2.x'
+ *          buildFile = file('gradle.properties')
+ *      }
+ *
+ * and then call it:
+ *
+ * ./gradlew performVersionUpgrade -PdependencyNewVersion=org.shipkit:shipkit:1.2.3
```

Comment:
So far, we have been using short names for project properties ("dryRun", "preview") for ease of use. I suggest to call this project property in similar fashion, e.g. "dependency".